phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -3322,7 +3331,7 @@ public Void call() throws Exception {
                         } finally {
                             try {
                                 if (!success && hConnectionEstablished) {
-                                    connection.close();
+                                    closeConnection();
```

Comment:
One nice side-effect of your change is that we will be decrementing `GLOBAL_HCONNECTIONS_COUNTER` at this point now, which looks like we were missing before.

Reply:
yes, @ChinmaySKulkarni I noticed that too. decrement was not good earlier